changelog/12191.bugfix.rst

```diff
@@ -0,0 +1 @@
+Finally fixed the issue where keyboard interrupts and system exits were not being respected during the tests collection phase in pytest! No more frustration of tests not ending correctly — happy testing! 🐞🔨🎉
```

Comment:
I like this message, but to keep in tone with the rest of our change log messages:

```suggestion
Keyboard interrupts and system exits are now probably handled during the test collection.
```